nimare/__init__.py

```diff
@@ -7,9 +7,6 @@
     warnings.simplefilter("ignore")
     from .dataset import Dataset
     from .meta import cbma
-    from .annotate import text
-    from .decode import gclda_decode_roi
-    from .parcellate import CoordCBP
     from .version import __version__
 
-    del cbma, Dataset, text, gclda_decode_roi, CoordCBP
+    del cbma, Dataset
```

Comment:
This is a _really_ bad way I've set up my init files in the past. I want to be able to access the submodule from the top level (e.g., `import nimare; nimare.meta`), which means I need to import the submodule in the init file, but I don't want to have some random function or class polluting the top level (e.g., `import nimare; nimare.CoordCBP`), so I delete it.

I've never come across a better way to do it, but would be happy to use another approach.

Reply:
You don't need to explicitly import submodules. If you're using `find_packages()`, and have an `__init__.py` file in all the modules you want to make available, you should be fine without the imports.

Reply:
We are using `find_packages` in `setup.py`, but what I can access from the top level (`nimare.` rather than `from nimare import`) is still dependent on the init file.

So if I don't have an submodule imports in the init file, typing in `nimare.` and hitting tab gives me:
<img width="235" alt="screen shot 2019-01-17 at 2 33 53 pm" src="https://user-images.githubusercontent.com/8228902/51343950-8a8c9980-1a65-11e9-93fe-83f6f422a219.png">

If I have the submodule imports, but don't delete the objects imported from those submodules, I get:
![image](https://user-images.githubusercontent.com/8228902/51343994-a42de100-1a65-11e9-92c8-a6642aa608bd.png)
I get why they're there, but I don't want random objects from the submodules like `Dataset` and  `text` showing up there.

Finally, if I import and delete, I get:
![image](https://user-images.githubusercontent.com/8228902/51344037-c4f63680-1a65-11e9-8136-aa876a2b211a.png)
Which just looks cleaner to me.

That said, if this isn't a good way to write init files, I can change things.

Reply:
I'm saying that you should just delete the submodule imports. There's no need for them. You should still be able to do `from nimare.annotate.blah import Blah`.

Reply:
(If that doesn't work, it's because something is probably not configured properly in setup or in the submodule `__init__.py` files. Also, you can control what's publicly displayed by setting an `__all__` in each submodule.)

Reply:
I can do that. I like having the tab-completion from the top level but I get why that's not a good way to set it up.

Reply:
One other thing: if you're looking at tab completions in IPython, keep in mind that IPython has its own rules for displaying possible completions; it doesn't respect the Python convention of only publicly showing names in `__all__` (see https://github.com/ipython/ipykernel/issues/129). So it's possible you can't have exactly what you want in terms of tab completion in IPython/Jupyter without some extra configuration, but I don't think that should be our concern here; the relevant considerations are (a) can people directly import from any submodule you want them to be able to import from, and (b) what's actually in the public namespace of the root-level module.

Reply:
Just to clarify, I'm not averse to importing the submodules. I agree that it's useful to be able to see what's available in the root namespace. It's the `del` statement I think is unnecessary. If you don't want to see things like `.text`, you can configure IPython locally to hide variables not in `__all__` from tab completion.

One other thing to keep in mind is that if the submodule imports are included, `import nimare` becomes pretty slow, because some of those submodules pull in heavy dependencies as they're being scanned. The optimal solution is probably to isolate the heavy imports (either by not importing submodules with big dependencies, or by deferring those imports within the submodules). That said, I don't think this is worth worrying about at this stage—an extra second or two is not going to kill anyone when a typical meta-analysis takes much longer than that.

Reply:
I never realized that it could be something I could configure in IPython. That's pretty cool.

I'm happy to go with the streamlined version for now, and then I can do a better job of specifying `__all__` and importing relevant things later.

Reply:
I think I misunderstood what you were trying to do... if you want to import the submodule, without any of its members, you can do `from . import annotate`. No need to `from .annotate import text` to get the submodule to load indirectly. That will add only the submodules to the namespace, without bringing any of its members into the root.

Reply:
🤦‍♂️ ... I've never tried that... I only ever tried `import annotate`, which obviously didn't work.

I think I'll switch to that in the init files. If later versions of NiMARE have enough stuff to seriously slow it down, I can switch over to the faster route then.

Reply:
No worries—Python packaging is not at all intuitive. I still Google this shit on a regular basis. :p